src/loader/aarch64/pe/mod.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2019 Intel Corporation. All rights reserved.
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE-BSD-3-Clause file.
+//
+// SPDX-License-Identifier: Apache-2.0 AND BSD-3-Clause
+
+//! Traits and structs for loading pe image kernels into guest memory.
+
+#![cfg(all(feature = "pe", target_arch = "aarch64"))]
+
+use std::error::{self, Error as StdError};
+use std::fmt::{self, Display};
+use std::io::{Read, Seek, SeekFrom};
+use std::mem;
+
+use vm_memory::{Address, ByteValued, Bytes, GuestAddress, GuestMemory, GuestUsize};
+
+use super::super::{Error as KernelLoaderError, KernelLoader, KernelLoaderResult, Result};
+
+/// ARM64 Image (PE) format support
+pub struct PE;
+
+unsafe impl ByteValued for arm64_image_header {}
+
+#[derive(Debug, PartialEq)]
+/// Elf kernel loader errors.
+pub enum Error {
+    /// Unable to seek to Image end.
+    SeekImageEnd,
+    /// Unable to seek to Image header.
+    SeekImageHeader,
+    /// Unable to read kernel image.
+    ReadKernelImage,
+    /// Unable to read Image header
+    ReadImageHeader,
+    /// Invalid Image binary.
+    InvalidImage,
+    /// Invalid Image magic number.
+    InvalidImageMagicNumber,
+}
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        match self {
+            Error::SeekImageEnd => "Unable to seek Image end",
+            Error::SeekImageHeader => "Unable to seek image header",
```

Comment:
Please keep `image` spelling consistent (either with capital I everywhere, or not)